File: core/templates/dev/head/video_list/VideoListService.js
// Copyright 2015 The Oppia Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// You may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS-IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * @fileoverview Translation functions for Oppia.
 *
 * @author milagro.teruel@gmail.com (Milagro Teruel)
 */

// Translations of strings that are loaded in the front page. They are listed
oppia.factory('videoListService', ['$http',
  function($http) {
    var _VIDEO_LIST_DATA_HANLDER_URL = '/video/data/' + categoryId + '/';
    return {
      getVideoList: function() {
        return $http.get(_VIDEO_LIST_DATA_HANLDER_URL + '0');
      },
      createVideo: function(name, ids, category) {
        return $http.post(_VIDEO_LIST_DATA_HANLDER_URL + '0', {
          name: name,
          ids: ids,
          category: category
        });
      },
      saveEditVideo: function(id, name, ids, category) {
        return $http.post(_VIDEO_LIST_DATA_HANLDER_URL + id, {
          id: id,
          name: name,
          ids: ids,
          category: category
        });
      },
      getVideo: function(id, $scope) {
        $http.get(_VIDEO_LIST_DATA_HANLDER_URL + id).then(function(response) {
          var data = response.data;
          $scope.name = data.name;
          $scope.id = data.id;
          $scope.ids = data.ids;
          $scope.create_on = data.create_on;
          $scope.category = data.category;
        });
      },
      deleteData: function(id) {
        return $http.delete(_VIDEO_LIST_DATA_HANLDER_URL + id);
      }
    };
  }]);

File: core/domain/video_list_service.py
# coding: utf-8
#
# Copyright 2014 The Oppia Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Commands for video list operations."""

from core.domain import video_list_demain
# from core.platform import models
from core.storage.video_list import gae_models as video_list_models
import feconf


def create_video(author_id, name, category, mid):
    """
    创建一个视频
    :param author_id:创建人id
    :param name:视频名称
    :param category:分类
    :param mid:对应的地址
    :return:None
    """

    video = video_list_models.VideoList()
    video.name = name
    video.category = category
    video.author_id = author_id
    video.ids = mid
    video.put()


def _get_video_from_model(model):
    return video_list_demain \
        .VideoList(model.id,
                   model.name,
                   model.ids,
                   model.category,
                   model.author_id,
                   model.created_on,
                   model.last_updated
                  )


def get_all_video(page_size=feconf.FEEDBACK_TAB_PAGE_SIZE,
                  urlsafe_start_cursor=None):
    results, new_urlsafe_start_cursor, more = \
        video_list_models.VideoList.get_all_video(page_size,
                                                  urlsafe_start_cursor)
    return [_get_video_from_model(model)
            for model in results], \
           new_urlsafe_start_cursor, more


def get_by_id(vid):
    return _get_video_from_model(
        video_list_models.VideoList.get(long(vid)))


def delete_video(vid):
    video_list_models.VideoList.get(long(vid)).delete()
